Feat(mesh): Requirements (#3843)
Kong/developer.konghq.com: app/mesh/resource-sizing-guidelines.md

    url: /mesh/concepts/
---

## Sizing your control plane

Generally, a control plane with 4vCPU and 2GB of memory will be able to accommodate more than 1000 data planes.

While it's difficult to give a precise number, a good rule of thumb is to assign about 1MB of memory per data plane.
When it comes to CPUs, {{site.mesh_product_name}} handles parallelism extremely well since its architecture uses a lot of shared nothing go-routines, so more CPUs usually enable quicker propagation of changes.

However, we highly recommend that you to run your own load tests prior to going to production.
There are many ways to run workloads and deploy applications, and while we test some of them, you are in the best position to build a realistic benchmark of what you do.

To see if you may need to increase your control plane's spec, there are two main metrics to pay attention to:

- Propagation time (`xds_delivery`): This is the time it takes between a change in the mesh and the data plane receiving its updated configuration. Think about it as the reactivity of your mesh.
- Configuration generation time (`xds_generation`): This is the time it takes for the configuration to be generated.

> For any large mesh using a transparent proxy it's highly recommended to use [reachable services](/mesh/performance-tuning/#reachable-services).

Reviewer: ```suggestion
For any large mesh using a transparent proxy, we recommend using [reachable services](/mesh/performance-tuning/#reachable-services).
```

You can also find tuning configuration in the [fine-tuning](/mesh/performance-tuning/) section of the docs.

## Sizing your sidecar container on Kubernetes

When deploying {{site.mesh_product_name}} on Kubernetes, the sidecar is deployed as a separate container, `kuma-sidecar`, in your pods. By default it has the following resource requests and limits:

```yaml
resources:
    requests:
        cpu: 50m
        memory: 64Mi
    limits:
        cpu: 1000m
        memory: 512Mi
```

> This configuration should be enough for most use cases. In some cases, when you can't scale horizontally or your service handles lots of concurrent traffic for example, you may need to change these values. You can do this using the [`ContainerPatch` resource](/mesh/data-plane-kubernetes/#custom-container-configuration). 

Reviewer: ```suggestion
This configuration should be enough for most use cases. In some cases (for example, when you can't scale horizontally or your service handles lots of concurrent traffic), you may need to change these values. You can do this using the [`ContainerPatch` resource](/mesh/data-plane-kubernetes/#custom-container-configuration). 
```
